File: 2- Elementary Sorts/Shell Sort/Shell.cpp
#include <vector>
using namespace std;

class Shell{
public:
    template <typename T>
    static void Sort(vector<T> &it){
        for (int h = it.size()/2; h > 0; h /= 2)
        {
            for (int i = h; i < it.size(); i += 1)
            {
                T temp = it[i];
                int j;
                for (j = i; j >= h && it[j - h]>temp; j -= h)
                    it[j] = it[j - h];
                it[j] = temp;
            }
        }
    }
};

File: 2- Elementary Sorts/Shell Sort/Shell.py
class Shell:
    @staticmethod
    def Sort(iterable):
        h = len(iterable) // 2
        while h > 0:
            i = 0
            j = h
            while j < len(iterable):
                if iterable[i] > iterable[j]:
                    iterable[i], iterable[j] = iterable[j], iterable[i]
                i += 1
                j += 1
                k = i
                while k - h > -1:
                    if iterable[k - h] > iterable[k]:
                        iterable[k - h], iterable[k] = iterable[k], iterable[k - h]
                    k -= 1
            h //= 2